google/cloud/documentai_toolbox/wrappers/hocr.py

```diff
@@ -0,0 +1,241 @@
+# uncompyle6 version 3.9.0
+# Python bytecode version base 3.8.0 (3413)
+# Decompiled from: Python 3.8.13 (default, Aug  2 2022, 12:17:31) 
+# [Clang 13.1.6 (clang-1316.0.21.2.5)]
+# Embedded file name: /Users/galzahavi/Documents/GitHub/gal/python-documentai-toolbox/google/cloud/documentai_toolbox/wrappers/hocr_document.py
+# Compiled at: 2023-04-05 14:28:21
+# Size of source mod 2**32: 10917 bytes
+import dataclasses
+from typing import List, Union
+from google.cloud import documentai
+
+ElementWithLayout = Union[(
+ documentai.Document.Page.Paragraph,
+ documentai.Document.Page,
+ documentai.Document.Page.Token,
+ documentai.Document.Page.Block,
+ documentai.Document.Page.Line)]
+
+def _get_bounding_box(object: ElementWithLayout, dimensions: documentai.Document.Page.Dimension):
+    if object.layout.bounding_poly.vertices:
+        min_x, min_y = object.layout.bounding_poly.vertices[0].x, object.layout.bounding_poly.vertices[0].y
+        max_x, max_y = object.layout.bounding_poly.vertices[2].x, object.layout.bounding_poly.vertices[2].y
+        return f"bbox {int(min_x)} {int(min_y)} {int(max_x)} {int(max_y)}"
+    min_x, min_y = object.layout.bounding_poly.normalized_vertices[0].x, object.layout.bounding_poly.normalized_vertices[0].y
+    max_x, max_y = object.layout.bounding_poly.normalized_vertices[2].x, object.layout.bounding_poly.normalized_vertices[2].y
+    return f"bbox {int(min_x * dimensions.width)} {int(min_y * dimensions.height)} {int(max_x * dimensions.width)} {int(max_y * dimensions.height)}"
+
+
+def _get_text(object: ElementWithLayout, document_text):
+    start_index = object.layout.text_anchor.text_segments[0].start_index
+    end_index = object.layout.text_anchor.text_segments[0].end_index
+    return document_text[start_index:end_index].replace('/n', '')
+
+
+def _load_hocr_words(words: List[documentai.Document.Page.Token], document_text: str, dimensions: documentai.Document.Page.Dimension):
+    w = []
+    for word in words:
+        text = _get_text(word, document_text)
+        w.append(hOCR_Word(text=text, documentai_word=word, dimensions=dimensions))
+    else:
+        return w
+
+
+def _load_hocr_lines(lines: List[documentai.Document.Page.Line], page: documentai.Document.Page, document_text: str):
+    l = []
+    for line in lines:
+        start_index = line.layout.text_anchor.text_segments[0].start_index
+        end_index = line.layout.text_anchor.text_segments[0].end_index
+        words = [word for word in page.tokens if word.layout.text_anchor.text_segments[0].start_index >= start_index if word.layout.text_anchor.text_segments[0].end_index <= end_index]
```

Comment:
If we put the `_load_hocr_lines,_load_hocr_words,etc...` logic in the respective toolbox wrapper, it would make instantiation super slow. The approach 2 implementation is really slow compared to this implementation.

Reply:
The Toolbox is the reasonable place to implement that logic (which we are doing here, but doing it in a way to be used only for hOCR export, while it is a more general concept that belongs to the Document AI Documents themselves).  Should we not expect the user to want to iterate over all the lines on a Toolbox-wrapped Page, and do something about the lines?